src/NUnitFramework/framework/Internal/AsyncToSyncAdapter.cs

```diff
@@ -43,6 +43,24 @@ public static bool IsAsyncOperation(Delegate @delegate)
             return IsAsyncOperation(@delegate.GetMethodInfo());
         }
 
+#if !THREAD_ABORT
```

Comment:
I wouldn't restrict it for Core only.
It is possible that my tests will have multiple TargetFrameworks - for both .NET Framework and .NET Core. I would prefer to be able to have CancellationToken parameter, even if it no-op for .NET Framework.

Reply:
For .NET Framework we can do the same and potentially drop the `Thread.Abort` in case the test method expects a `CancellationToken` or turn this into a two prong approach. Do `CancellationToken` first so _behaving_ code works, followed by a `Thread.Abort` a bit later.

Reply:
It may just be that I haven't had my coffee yet, but I'd like to be clear on the suggestion here.

If I understand your suggestion here @manfred-brands , are you saying we should always allow an extra `CancellationToken` param to a test method and only use cooperative cancellation on netstandard builds of the framework? And that a follow-up change could use `Thread.Abort` to cancel tests on netfx builds of the framework?

If so, I agree with that approach.

Reply:
Actually I'm curious about how it could behave if a netstandard build were used in a netfx context

Reply:
> Actually I'm curious about how it could behave if a netstandard build were used in a netfx context

NuGet will pick the netfx build for a netfx context.
As said I would rather have a `CancelAfterAttribute` working for both and mark the `TimeoutAttribute` as Framework only.

Reply:
Ok, that's good about NuGet. I'd been concerned that some historical upgrade paths could inadvertently pull in a netstandard build but it sounds like it's not possible.